R/updateNullModelApproxScoreSE.R

```diff
@@ -0,0 +1,246 @@
+
+setMethod(".scoreSEtable",
+          "SeqVarData",
+          function(gdsobj, null.model, variant.id,
+                    sparse=TRUE, imputed=FALSE, male.diploid=TRUE, genome.build=c("hg19", "hg38"), verbose = TRUE){
+
+               # don't use sparse matrices for imputed dosages
+               if (imputed) sparse <- FALSE
+
+               # coerce null.model if necessary
+               if (sparse) null.model <- .nullModelAsMatrix(null.model)
+
+               # filter samples to match null model
+               sample.index <- .setFilterNullModel(gdsobj, null.model, verbose=verbose)
+
+               # filter variants
+               seqSetFilter(gdsobj, variant.id = variant.id, verbose = verbose)
+
+               # check ploidy
+               if (SeqVarTools:::.ploidy(gdsobj) == 1) male.diploid <- FALSE
+
+               # variant info
+               var.info <- variantInfo(gdsobj, alleles=FALSE, expanded=TRUE)
+
+               # read in genotype data
+               if (!imputed) {
+                    geno <- expandedAltDosage(gdsobj, use.names=FALSE, sparse=sparse)[sample.index,,drop=FALSE]
+               } else {
+                    geno <- imputedDosage(gdsobj, use.names=FALSE)[sample.index,,drop=FALSE]
+               }
+
+               # take note of number of non-missing samples
+               n.obs <- .countNonMissing(geno, MARGIN = 2)
+
+               # allele frequency
+               freq <- .alleleFreq(gdsobj, geno, sample.index=sample.index,
+                                        male.diploid=male.diploid, genome.build=genome.build)
+
+               # filter monomorphic variants
+               keep <- .filterMonomorphic(geno, count=n.obs, freq=freq$freq, imputed=imputed)
+               if (!all(keep)) {
+                    var.info <- var.info[keep,,drop=FALSE]
+                    geno <- geno[,keep,drop=FALSE]
+                    n.obs <- n.obs[keep]
+                    freq <- freq[keep,,drop=FALSE]
+               }
+
+               # mean impute missing values
+               if (any(n.obs < nrow(geno))) {
+                    geno <- .meanImpute(geno, freq$freq)
+               }
+
+               geno <- .genoAsMatrix(null.model, geno)
+
+               # true variance
+               Gtilde <- calcGtilde(null.model, geno)
+               se.true <- sqrt(colSums(Gtilde^2)) # sqrt(GPG)
+
+               # approx variance
+               Gtilde <- calcGtildeApprox(null.model, geno, r = 1)
+               se.approx <- sqrt(colSums(Gtilde^2)) #sqrt(GWG)
+
+               # results
+               tab <- cbind(var.info, n.obs, freq, score.SE.true = se.true, score.SE.approx = se.approx, score.SE.ratio = se.true/se.approx)
+               return(tab)
+
+          })
+
+setMethod(".scoreSEtable",
+          "GenotypeData",
+          function(gdsobj, null.model, variant.id, male.diploid=TRUE, verbose = TRUE){
+
+               stop("function not implemented for GenotypeData objects")
+
```

Comment:
Same here. I think we just drop this commented out code if we decide it's unnecessary to implement for now.

Reply:
You don't even need to define a method with an error message, because if you try to call a generic function on an object that doesn't have a method implemented, you automatically get an error message like "no S4 method for this object type"